stl/inc/ranges

```diff
@@ -22,15 +25,462 @@ _STL_DISABLE_CLANG_WARNINGS
 
 _STD_BEGIN
 namespace ranges {
-    // Much machinery defined in <xutility>
+    // MUCH machinery defined in <xutility>
 
     // clang-format off
     // CONCEPT ranges::viewable_range
     template <class _Rng>
     concept viewable_range = range<_Rng>
         && (borrowed_range<_Rng> || view<remove_cvref_t<_Rng>>);
+
+    // CLASS TEMPLATE ranges::_Semiregular_box
+    // clang-format off
+    template <class _Ty>
+    concept _Copy_constructible_object = copy_constructible<_Ty> && is_object_v<_Ty>;
+    // clang-format on
+
+#if 0 // TRANSITION, VSO-1174090

```

Comment:
Should we activate this workaround for C1XX only? It appears to be unusually large and potentially ABI-impacting (which is fine for C++20 code, just unusual).

Reply:
I made the workaround unconditional _because_ the two branches are ABI-incompatible. If for whatever reason this gets released as-is we can't have link-incompatible clang and MSVC TUs. That said, it's almost certain the workaround will be removed before this code is released since the MSVC bug is already fixed internally.